### PR TITLE
:seedling: Generic arm64 isos

### LINKED ISSUE #790: :penguin: Generic arm64 images
**Is your feature request related to a problem? Please describe.**
We do have HW flavored images, however, there are cases when generic arm64 images, even with ISOs are a good fit too: that could work both for VMs on Macs, or HW generic arm hardware which support is streamlined

**Describe the solution you'd like**
Generic arm64 flavors, generating ISOs and such

**Describe alternatives you've considered**
Keep only dedicated images

### PR DESCRIPTION
**Pull request by Itxaka**
**What this PR does / why we need it**:

Generates arm64 generic isos able to run under qemu emulation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #790

### CODE REVIEW

**Comment by Itxaka** [comment]
Not ready, base url should be changed back to the quay default and the iso being generated should run only on master and artifact saved so it can be downloaded and tested.

Just want to see if I broke something else meanwhile and how the latest os builder image works

**Comment by Itxaka** [comment]
```
localhost:~ # uname -i
aarch64
localhost:~ # cat /etc/os-release 
NAME="kairos-core-opensuse-leap"
VERSION="v2.0.0-rc6-6-gf2a6d77-dirty"
ID="kairos"
ID_LIKE="kairos-core-opensuse-leap"
VERSION_ID="v2.0.0-rc6-6-gf2a6d77-dirty"
PRETTY_NAME="kairos-core-opensuse-leap v2.0.0-rc6-6-gf2a6d77-dirty"
ANSI_COLOR="0;32"
BUG_REPORT_URL="https://github.com/kairos-io/kairos/issues"
HOME_URL="https://github.com/kairos-io/kairos"
IMAGE_REPO="quay.io/kairos/core-opensuse-leap"
IMAGE_LABEL="latest"
GITHUB_REPO="kairos-io/kairos"
VARIANT="core"
FLAVOR="opensuse-leap"
```

nais

**Comment by Itxaka** [comment]
nasi, even k3s runs as expected so provider-kairos should work out of the box!

```
localhost:~ # /var/lib/k3s/k3s kubectl get nodes
NAME        STATUS   ROLES                  AGE   VERSION
localhost   Ready    control-plane,master   70s   v1.26.3+k3s1
```

File: Earthfile [hunk 1]
```diff
@@ -20,7 +21,7 @@ END
 ARG COSIGN_EXPERIMENTAL=0
 ARG CGO_ENABLED=0
 # renovate: datasource=docker depName=quay.io/kairos/osbuilder-tools versioning=semver-coerced
-ARG OSBUILDER_VERSION=v0.5.3
+ARG OSBUILDER_VERSION=v0.6.0
```
Comment (Itxaka) [inline Earthfile:24]: this version has support for multiarch packages for elemental and livecd grub artifacts. So using an arm64 image will have the proper artifacts for that arch on it.

File: .github/workflows/image-arm.yaml [hunk 2]
```diff
@@ -116,3 +116,35 @@ jobs:
...
           if-no-files-found: error
+  iso:
+    runs-on: ubuntu-latest
+    needs:
+      - get-matrix
+    strategy:
+      fail-fast: false
+      matrix:
+        flavor:
+          - "opensuse-leap"
```
Comment (Ludea) [inline .github/workflows/image-arm.yaml:127]: What about other flavor ? like Ubuntu, Debian, fedora or alpine ?

File: Earthfile [hunk 3]
```diff
@@ -362,15 +362,21 @@ base-image:
         RUN rm -rf /boot/initramfs-*
     END
 
+    # Set /boot/vmlinuz pointing to our kernel so elemental-cli can use it
+    # https://github.com/kairos-io/elemental-cli/blob/23ca64435fedb9f521c95e798d2c98d2714c53bd/pkg/elemental/elemental.go#L553
     IF [ ! -e "/boot/vmlinuz" ]
         # If it's an ARM flavor, we want a symlink here from zImage/Image
-        IF [ -e "/boot/Image" ]
+        # Check that its not a symlink already or grub will fail!
```
[the remark this answers is not in the document]
Comment (Itxaka) [reply Earthfile:369]: This seems to have broken debian 😂

[the remark this answers is not in the document]
Comment (Itxaka) [reply Earthfile:369]: nhow its fixed back and simpler

File: .github/workflows/image-arm.yaml [hunk 4]
```diff
@@ -116,3 +116,34 @@ jobs:
           name: ${{ matrix.flavor }}-image
           path: build
           if-no-files-found: error
+  iso:
```
Comment (mauromorales) [inline .github/workflows/image-arm.yaml:119]: do we plan to keep this one for PRs? or just for master pushes?

**Review by mauromorales — APPROVED** [review]
:muscle:

File: .github/workflows/image-arm.yaml [hunk 5]
```diff
@@ -116,3 +116,40 @@ jobs:
...
+        with:
+          platforms: all
+      - name: Set up Docker Buildx
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Build iso  🔧
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
+        run: earthly --platform=linux/arm64 +iso --FLAVOR=${{ matrix.flavor }}
+      - uses: actions/upload-artifact@v3
```
Comment (mudler) [inline .github/workflows/image-arm.yaml:149]: I guess we need to adapt as well the release pipeline?

**Review by mudler — APPROVED** [review]
Nais!

File: .github/workflows/image-arm.yaml (hunk 5 above)
Comment (Itxaka) [reply .github/workflows/image-arm.yaml:149]: I haven't touched release pipelines as I wanted to discuss on Monday what we need to support and how regarding arm64. Flavours, best-effort basis for arm64 issues, etc...

**Pull request merged**